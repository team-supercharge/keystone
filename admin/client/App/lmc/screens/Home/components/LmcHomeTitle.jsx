import React from 'react';


class LmcHomeTitle extends React.Component {
    render () {
        const { home } = this.props;
        return (
            <div style={styles.container}>
                <h2 style={styles.title}>
                    Hey 
                    <span style={styles.bold}> {Keystone.user.name}</span>
                    , welcome to the Care Office
                    {
                        home && home.name ?
                            <span> for <span style={styles.bold}>{ home.name }</span></span> 
                            : null
                    }
                    !
                </h2>
            </div>
        );
    }
};

const styles = {
    container: {
        padding: 0,
    },
    title: {
        fontSize: 20,
        color: '#e65d79',
    },
    bold: {
        fontWeight: 'bold',
    }
}

export default LmcHomeTitle;
