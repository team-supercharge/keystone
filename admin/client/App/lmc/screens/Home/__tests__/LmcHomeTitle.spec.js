import React from 'react'
import { shallow } from 'enzyme'
import MockDate from 'mockdate'
import LmcHomeTitle from '../components/LmcHomeTitle'

MockDate.set('1/1/2019')

describe('LmcHomeTitle', () => {
    let wrapper

    beforeAll(() => {
        global.Keystone = { 
            user: { 
                firstLogin: new Date('1/1/2019') 
            } 
        } 
    })

    beforeEach(() => {
        wrapper = shallow(
            <LmcHomeTitle
                residents={[]}
            />
        )
    })

    test('should render correct snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    afterAll(() => {
        MockDate.reset()
    })
})