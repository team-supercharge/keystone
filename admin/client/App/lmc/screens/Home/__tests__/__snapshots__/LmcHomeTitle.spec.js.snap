// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcHomeTitle should render correct snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "padding": 0,
      }
    }
  >
    <h2
      style={
        Object {
          "color": "#e65d79",
          "fontSize": 20,
        }
      }
    >
      <span
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Hey 
      </span>
      , 
      <span>
        welcome to the Care Office
        !
      </span>
    </h2>
    <p>
      This is where you manage your team, residents and the care provided in your home.​ To help you get started we’ve come up with a quick tour. This shouldn’t take more than a few minutes and at the end you’ll be a Rockstar when it comes to using it.​
      <br />
      <LmcTour
        forceOpen={false}
      />
    </p>
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <h2
        style={
          Object {
            "color": "#e65d79",
            "fontSize": 20,
          }
        }
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Hey 
        </span>
        , 
        <span>
          welcome to the Care Office
          !
        </span>
      </h2>
      <p>
        This is where you manage your team, residents and the care provided in your home.​ To help you get started we’ve come up with a quick tour. This shouldn’t take more than a few minutes and at the end you’ll be a Rockstar when it comes to using it.​
        <br />
        <LmcTour
          forceOpen={false}
        />
      </p>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcHomeTitle
        residents={Array []}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcHomeTitle {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "residents": Array [],
        },
        "refs": Object {},
        "renderHello": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "padding": 0,
            }
          }
        >
          <h2
            style={
              Object {
                "color": "#e65d79",
                "fontSize": 20,
              }
            }
          >
            <span
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Hey 
            </span>
            , 
            <span>
              welcome to the Care Office
              !
            </span>
          </h2>
          <p>
            This is where you manage your team, residents and the care provided in your home.​ To help you get started we’ve come up with a quick tour. This shouldn’t take more than a few minutes and at the end you’ll be a Rockstar when it comes to using it.​
            <br />
            <LmcTour
              forceOpen={false}
            />
          </p>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "padding": 0,
            }
          }
        >
          <h2
            style={
              Object {
                "color": "#e65d79",
                "fontSize": 20,
              }
            }
          >
            <span
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Hey 
            </span>
            , 
            <span>
              welcome to the Care Office
              !
            </span>
          </h2>
          <p>
            This is where you manage your team, residents and the care provided in your home.​ To help you get started we’ve come up with a quick tour. This shouldn’t take more than a few minutes and at the end you’ll be a Rockstar when it comes to using it.​
            <br />
            <LmcTour
              forceOpen={false}
            />
          </p>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcHomeTitle
    residents={Array []}
  />,
}
`;
